YOLayout/YOLayout_IMP.m

```diff
@@ -138,11 +138,11 @@ - (CGRect)setSize:(CGSize)size inRect:(CGRect)inRect view:(id)view options:(YOLa
     size = sizeThatFits;
   }
   
-  if ((options & YOLayoutOptionsSizeToFitHorizontal) == 0) {
+  if ((options & YOLayoutOptionsSizeToFitHorizontal) == 0 && originalSize.width > 0) {
```

Comment:
Yeah... this is a tough call...

So these are the scenarios:

Laying out vertically:
- A width is required to know the height (like multiline text). Specified width is > 0. If sizing returns a different width it's overriden.
- A width is not required to know the height (like a button). Specified width is 0. Button decides height. YOView can have 0 width and height.

Laying out horizontally:
- A height is required to know the width (like some grid thing?). Specified width is > 0. If sizing returns a different width it's overriden.
- A height is not required to know the width (like a button). Specified width is 0. Button decides height. YOView can have 0 width and height.

Checking the originalSize is non-zero before overriding is compatible with all these scenarios so I think its ok?